packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -812,6 +859,63 @@ export class Project extends ProjectBase {
       : this._secondaryArtifacts.map((secondaryArtifact) => secondaryArtifact.toArtifactsJSON());
   }
 
+  /**
+   * If configured, set up the VPC-related properties
+   *
+   * Returns the VpcConfig that should be added to the
+   * Lambda creation properties.
+   */
+  private configureVpc(props: ProjectProps): CfnProject.VpcConfigProperty | undefined {
+    if ((props.securityGroups || props.allowAllOutbound !== undefined) && !props.vpc) {
+      throw new Error(`Cannot configure 'securityGroup' or 'allowAllOutbound' without configuring a VPC`);
+    }
+
+    if (!props.vpc) { return undefined; }
+
+    if (props.securityGroups && props.allowAllOutbound !== undefined) {
+      throw new Error(`Configure 'allowAllOutbound' directly on the supplied SecurityGroup.`);
+    }
+
+    if (props.securityGroups && props.securityGroups.length > 0) {
+      this.securityGroups = [...this.securityGroups, ...props.securityGroups];
+    } else {
+      const securityGroup = new ec2.SecurityGroup(this, 'SecurityGroup', {
+        vpc: props.vpc,
+        description: 'Automatic generated security group for CodeBuild ' + this.node.uniqueId,
+        allowAllOutbound: props.allowAllOutbound
+      });
+      this.securityGroups.push(securityGroup);
+    }
+    const securityGroupIDs = this.securityGroups.map(s => s.securityGroupId);
+    const vpcPlacement: ec2.VpcPlacementStrategy = props.vpcPlacement ? props.vpcPlacement : {
+      subnetsToUse: ec2.SubnetType.Private
+    };
+    const subnets = props.vpc.subnets(vpcPlacement);
+    const policyStatement = new iam.PolicyStatement(iam.PolicyStatementEffect.Allow);
+    policyStatement.addAllResources();
+    policyStatement.addActions(
+      'ec2:CreateNetworkInterface',
+      'ec2:DescribeNetworkInterfaces',
+      'ec2:DeleteNetworkInterface',
+      'ec2:DescribeSubnets',
+      'ec2:DescribeSecurityGroups',
+      'ec2:DescribeDhcpOptions',
+      'ec2:DescribeVpcs',
+      'ec2:CreateNetworkInterfacePermission'
+    );
+    this.attachInlineRolePolicy(new iam.Policy(this, 'CodeBuildPolicy', {
```

Comment:
Use the `this.addToRolePolicy` method instead. This way, you'll be able to get rid of the `attachInlineRolePolicy` method.